src/glum/_glm.py

```diff
@@ -718,6 +718,59 @@ def link_instance(self) -> Link:
         else:
             return get_link(self.link, self.family_instance)
 
+    @property
+    def aic(self):
+        """
+        Akaike's information criteria. Computed as:
+        :math:`-2\\log\\hat{\\mathcal{L}} + 2\\hat{k}` where
+        :math:`\\hat{\\mathcal{L}}` is the maximum likelihood estimate of the
+        model, and :math:`\\hat{k}` is the effective number of parameters under
+        an assumption of lasso regularisation. In practice, this is the
+        number of non-zero coefficients. When ridge or elastic net
+        regularisation is used, this attribute returns a warning.
+        """
+        return self.get_info_criteria("aic")
+
+    @property
+    def aicc(self):
+        """
+        Second-order Akaike's information criteria (or small sample AIC).
+        Computed as:
+        :math:` -2\\log\\hat{\\mathcal{L}} + 2\\hat{k} + \\frac{2k(k+1)}{n-k-1}`
+        where :math:`\\hat{\\mathcal{L}}` is the maximum likelihood estimate of
+        the model, :math:`n` is the number of training instances, and
+        :math:`\\hat{k}` is the effective number of parameters under an
+        assumption of lasso regularisation. In practice, this is the number of
+        non-zero coefficients. When ridge or elastic net regularisation is used,
+        this attribute returns a warning.
+        """
+        return self.get_info_criteria("aicc")
+
+    @property
+    def bic(self):
+        """
+        Bayesian information criterion. Computed as:
+        :math:`-2\\log\\hat{\\mathcal{L}} + k\\log(n)` where
+        :math:`\\hat{\\mathcal{L}}` is the maximum likelihood estimate of the
+        model, :math:`n` is the number of training instances, and
+        :math:`\\hat{k}` is the effective number of parameters under an
+        assumption of lasso regularisation. In practice, this is the number of
+        non-zero coefficients. When ridge or elastic net regularisation is used,
+        this attribute returns a warning.
+        """
+        return self.get_info_criteria("bic")
+
+    def get_info_criteria(self, crit: str):
+        if hasattr(self, "_info_criteria"):
+            if (self.alpha is not None and self.alpha > 0) or self.l1_ratio < 1.0:
+                warnings.warn(
+                    "There is no robust definition for the model's degrees of "
+                    + f"freedom under L2 (ridge) regularisation. The {crit} "
+                    + "might not be well defined in these cases."
+                )
+            return self._info_criteria[crit]
+        return None
```

Comment:
I think it would make more sense to raise an error if the model as not been fitted before (so that we don't have the `_info_criteria` attribute). This would be similar to trying to predict using a model that has not been fitted.